### PR TITLE
Remove some unused method parameters

### PR DESCRIPTION
**Pull request by lacostej**
I made a first pass at running rubocop on unused method parameters and found these. Also a little code indentation cleanup.

All changes are in individual commits.

More improvements on the same topic coming later.

### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.
- [x] I've added or updated relevant unit tests.

### CODE REVIEW

**Review by rogerluan — APPROVED** [review]
LGTM, just left a comment regarding a seemingly outdated comment, and potentially unnecessary changes to the `Gemfile.lock` 😊 

I always like cleaning up dead code 😁

File: Gemfile.lock
[the remark this answers is not in the document]
Comment (lacostej) [reply Gemfile.lock:1]: I did change it to test the latest version of rubocop before I started playing with it.

Not aware of the process for changing the Gemfile.lock during releases. Need to look into it.

File: deliver/lib/deliver/upload_screenshots.rb [hunk 1]
```diff
@@ -183,7 +180,7 @@ def wait_for_complete(iterator)
     end
 
     # Verify all screenshots states on App Store Connect are okay
-    def retry_upload_screenshots_if_needed(iterator, states, number_of_screenshots, tries, localizations, screenshots_per_language)
```
[the remark this answers is not in the document]
Comment (lacostej) [reply deliver/lib/deliver/upload_screenshots.rb:186]: I can drop that particular change for now.

**Review by rogerluan — APPROVED** [review]
🚀

**Pull request merged**